app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/CreateAccountActivityTest.java

```diff
@@ -0,0 +1,462 @@
+
+        usernameText = (EditText) solo.getView(R.id.create_username);
+        emailText = (EditText) solo.getView(R.id.create_email);
+        phoneNumberText = (EditText) solo.getView(R.id.create_phone_number);
+        password1Text = (EditText) solo.getView(R.id.create_password);
+        password2Text = (EditText) solo.getView(R.id.confirm_password);
+
+        fb = FirebaseFirestore.getInstance();
+        colRef = fb.collection(resources.getString(R.string.users_collection));
+    }
+
+    /**
+     * Gets the activity
+     *
+     * @throws Exception
+     */
+    @Test
+    public void start() throws Exception {
+        Activity activity = rule.getActivity();
+    }
+
+    /**
+     * Tests create account button functionality with no input
+     */
+    @Test
+    public void emptyCreateAccountTest()
+    {
+        solo.clickOnButton(resources.getString(R.string.create_account));
+
+        Assert.assertTrue(solo.waitForText(EditTextValidator.EMPTY));
+
+        Assert.assertNotNull(usernameText.getError());
+        Assert.assertNotNull(emailText.getError());
+        Assert.assertNotNull(phoneNumberText.getError());
+        Assert.assertNotNull(password1Text.getError());
+        Assert.assertNotNull(password2Text.getError());
+    }
+
+    /**
+     * Tests create account button functionality with spaces input
+     */
+    @Test
+    public void spacesCreateAccountTest()
+    {
+        solo.enterText(usernameText, resources.getString(R.string.space));
+        solo.enterText(emailText, resources.getString(R.string.space));
+        solo.enterText(phoneNumberText, resources.getString(R.string.space));
+        solo.enterText(password1Text, resources.getString(R.string.space));
+        solo.enterText(password2Text, resources.getString(R.string.space));
+
+        solo.clickOnButton(resources.getString(R.string.create_account));
+
+        Assert.assertTrue(solo.waitForText(EditTextValidator.EMPTY));
+
+        Assert.assertNotNull(usernameText.getError());
+        Assert.assertNotNull(emailText.getError());
+        Assert.assertNotNull(phoneNumberText.getError());
+        Assert.assertNotNull(password1Text.getError());
+        Assert.assertNotNull(password2Text.getError());
+    }
+
+    /**
+     * Tests for empty username and username with only space input
+     */
+    @Test
+    public void emptyAndSpaceUsernameTest()
+    {
+        solo.enterText(emailText, resources.getString(R.string.test_create_account_email));
+        solo.enterText(phoneNumberText, resources.getString(R.string.test_account1_phone));
+        solo.enterText(password1Text, resources.getString(R.string.test_account1_password));
+        solo.enterText(password2Text, resources.getString(R.string.test_account1_password));
+
+        solo.clickOnButton(resources.getString(R.string.create_account));
+
+        Assert.assertTrue(solo.waitForText(EditTextValidator.EMPTY));
+        Assert.assertNotNull(usernameText.getError());
```

Comment:
Can you add Assert.assertNull() to any fields that should not have errors? This will catch the case where we are erroneously showing errors in fields that are correctly filled.

Reply:
oooh. okay